### ISSUE TITLE
"no argument given for parameter" diagnostic incorrectly includes nonterminals in prose assertions

### ISSUE THREAD

**Issue by bakkot** [rbuckton/grammarkdown#52]
Consider the following grammar:

```
HexDigits[Sep] ::
  HexDigit
  HexDigits[?Sep] HexDigit
  [+Sep] HexDigits[?Sep] `_` HexDigit

HexDigit :: one of
  `0` `1` `2` `3` `4` `5` `6` `7` `8` `9` `a` `b` `c` `d` `e` `f` `A` `B` `C` `D` `E` `F`

NotCodePoint ::
  HexDigits[~Sep] [> but only if MV of |HexDigits| &gt; 0x10FFFF]
```

Grammarkdown has a diagnostic for the last line:

> error GM2007: There is no argument given for parameter 'Sep'.

but I wouldn't really expect to write the parameter in the `MV of |HexDigits|` part. Also the parser doesn't allow a parameter to occur in that position: if you change it to `MV of |HexDigits[~Sep]|`, then you get

> error GM2000: Cannot find name: 'HexDigits[~Sep]'.

**Comment by rbuckton** [comment]
This has been published as `grammarkdown@2.2.1`.

**Comment by bakkot** [comment]
Thanks!
